accepted/0000-sbom-command.md

```diff
@@ -0,0 +1,461 @@
+
+* There’s a case to be made that SBOM generation should just be a flag on the existing `npm-ls` command – an SBOM is really just a different format of the output already generated by `npm-ls` (in fact, the CycloneDX tool described above parses the output from `npm-ls` in order to generate its SBOM). However, many of the arguments supported by `npm-ls` don’t really make sense in the context of SBOM generation (package spec filtering, depth, etc…) so it feels like a bit of an awkward fit. \
+ \
+Making it a distinct command allows us to add SBOM-specific features in the future like a `--sign` flag which could be used to generate a signed SBOM. \
+ \
+_Recommendation: Add a distinct command for generating an SBOM._
+ 
+* Does `npm-sbom` command have a notion of a “default” SBOM format? Do we give preference to one of CycloneDX/SPDX or do we remain totally neutral (possibly at the expense of DX)? \
+ \
+_Recommendation: Default to SPDX if no format is specified._
+
+* Both CycloneDX and SPDX support multiple document formats (JSON, XML, Protocol Buffers, etc). Should we support output of multiple formats, or do we stick w/ JSON? \
+ \
+_Recommendation: Stick with JSON-only for the first version of this feature._
+
+
+## Example SBOMs
+
+The sections below show what the different SBOM formats would look like for a basic npm project with a handful of dependencies.
+
+The _package.json_ file below describes a “hello-world” application with two direct dependencies: the `debug` package and the `@tsconfig/node14` package (which is listed as a development dependency). The `debug` package itself has a dependency on the `ms` package.
+
+```json
+{
+  "name": "hello-world",
+  "version": "1.0.0",
+  "license": "Apache-2.0",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/example/hello-world.git"
+  },
+  "dependencies": {
+    "debug": "^4.3.0"
+  },
+  "devDependencies": {
+    "@tsconfig/node14": "^14.1.0"
+  }
+}
+```
+
+The complete dependency tree for this project looks like this:
+
+```
+$ npm ls
+
+hello-world@1.0.0 
+├── @tsconfig/node14@14.1.0
+└─┬ debug@4.3.4
+  └── ms@2.1.2
+```
+
+
+### CycloneDX
+
+The proposed CycloneDX SBOM generated for the project above would look like the following:
+
+```json
+{
+  "$schema": "https://cyclonedx.org/schema/bom-1.4.schema.json",
+  "bomFormat": "CycloneDX",
+  "specVersion": "1.4",
+  "serialNumber": "urn:uuid:f2fa9eae-72f1-430c-a9b3-986ffe05bc6e",
+  "version": 1,
+  "metadata": {
+    "timestamp": "2023-08-10T00:19:08.697Z",
+    "tools": [
+      {
+        "vendor": "npm",
+        "name": "cli",
+        "version": "9.8.1"
+      }
+    ],
+    "component": {
+      "bom-ref": "hello-world@1.0.0",
+      "type": "application",
+      "name": "hello-world",
+      "version": "1.0.0",
+      "purl": "pkg:npm/hello-world@1.0.0",
+      "properties": [
+        {
+          "name": "cdx:npm:package:path",
+          "value": ""
+        }
+      ]
+    }
+  },
+  "components": [
```

Comment:
`components.component` are missing the following important information: 
- licenses
- author
- description
- scope
- hashes
- externalReferences
  - distribution -- most importent, to know where a package was sourced from 
  - vcs
  - issue-tracker
  - website
  - documentation

see https://cyclonedx.org/docs/1.4/json/#components

Reply:
Updated RFC to include `scope` and `externalReferences.distribution`. Other fields are mentioned as being populated as-available.